src/external_references.rs

```diff
@@ -31,6 +33,14 @@ impl ExternalReferences {
   pub fn as_ptr(&self) -> *const intptr_t {
     self.null_terminated.as_ptr()
   }
+
+  pub fn push(&mut self, reference: ExternalReference) {
```

Comment:
Adding this API seems unnecessarily ad-hoc.
For testing purposes, I'd suggest to just build a `Vec<ExternalReference>` and convert that to a `ExternalReferences` object.